pretyping/inductiveops.ml

```diff
@@ -716,4 +716,6 @@ let control_only_guard env sigma c =
     check_fix_cofix env c;
     EConstr.iter_with_full_binders env sigma EConstr.push_rel iter env c
   in
-  iter env c
+  try iter env c
+  with Type_errors.TypeError (env, e) ->
+    Loc.raise (Pretype_errors.PretypeError (env, sigma, TypingError (Type_errors.map_ptype_error EConstr.of_constr e)))
```

Comment:
Why Loc.raise? IMO should be either Exninfo.raise with an info captured from the kernel exception, or unqualified raise

Reply:
> Why Loc.raise?

I don't know. I just copied what was done for `Pretyping.esearch_guard`.

I'd be ok with `Exninfo.raise` too. Should we then change also `Pretyping.esearch_guard`?

Reply:
Unqualified raise is probably better, we don't really care about the backtrace at this point.

Reply:
Yes.